perfkitbenchmarker/windows_virtual_machine.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+from perfkitbenchmarker import disk
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import virtual_machine
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker import windows_packages
+
+
+FLAGS = flags.FLAGS
+
+STARTUP_SCRIPT = ('powershell -Command "Enable-PSRemoting -force; '
+                  'Set-Item wsman:\\localhost\\client\\trustedhosts * -Force; '
+                  'Restart-Service WinRM; '
+                  'netsh advfirewall firewall add rule name=\'Port 5985\' '
+                  'dir=in action=allow protocol=TCP localport=5985"')
+DEFAULT_WINRM_PORT = 5985
```

Comment:
Hm, next problem. WindowsGceVirtualMachine uses windows_virtual_machine.STARTUP_SCRIPT which will always contain the hardcoded DEFAULT_WINRM_PORT. If a subclass were to override self.winrm_port, they'd get out of sync again.

Consider either moving the startup script generation to a method, so that it can use the current self attribute value, or removing the DEFAULT_ prefix if it's unlikely that people will want to change it.